### PR TITLE
[12.0][IMP] hr_timesheet_sheet: more extendability & cleanup

### PR DESCRIPTION
**Pull request by alexey-pelykh**
* Remove after-migration remains
 * Refactor some methods to be more extendable (for #154)

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
@bjeficent @aheficent you're aware of the topic, so please review this PR

**Comment by alexey-pelykh** [comment]
@pedrobaeza do we need to bump version somehow in this case?

**Review by bjeficent — APPROVED** [review]
Functional review :+1:

**Comment by alexey-pelykh** [comment]
@tarteo if you're aware of the topic - please review this one as well

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM :+1:

File: hr_timesheet_sheet/__manifest__.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
     'name': 'HR Timesheet Sheet',
-    'version': '12.0.1.0.0',
+    'version': '12.0.1.0.1',
     'category': 'Human Resources',
     'sequence': 80,
     'summary': 'Timesheet Sheets, Activities',
     'license': 'AGPL-3',
     'author': (
         'Eficent, '
+        'Brainbean Apps, '
```
Comment (pedrobaeza) [inline hr_timesheet_sheet/__manifest__.py]: This change doesn't grant you co-authorship.

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 2]
```diff
@@ -296,7 +287,7 @@ def onchange_add_project_id(self):
             }
 
     @api.multi
-    def copy(self, default=None):
+    def copy(self, default=None):  # pylint: disable=method-required-super
```
Comment (pedrobaeza) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: Don't do that, but use a possible "exit" scheme:
```python
if self.env.context.get('allow_copy_timesheet'):
    return super().copy(default=default)
raise UserError(...)
```

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 3]
```diff
@@ -446,10 +471,10 @@ def _prepare_empty_analytic_line(self):
             'name': '/',
             'employee_id': self.employee_id.id,
             'date': self.date_start,
-            'project_id': self.add_line_project_id and
-            self.add_line_project_id.id or False,
-            'task_id': self.add_line_task_id and
-            self.add_line_task_id.id or False,
+            'project_id': self.add_line_project_id.id
+                    if self.add_line_project_id else False,
```
Comment (pedrobaeza) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]: This is not correct: you should enclose full expression after `:` into parenthesis.

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 4]
```diff
@@ -651,3 +667,15 @@ def _line_to_timesheet(self, amount):
             'unit_amount': amount,
             'company_id': self.sheet_id.company_id.id,
         }
+
+    @api.model
+    def _find_timesheets(self):
+        task = self.task_id and self.task_id.id or False
```
Comment (pedrobaeza) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py]:
```suggestion
        task_id = self.task_id.id
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py (hunk 2 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py]: Don't do what? Disable linter if no super call intended?

Comment (alexey-pelykh) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py]: Yet your suggestion is great, this would allow extend this behaviour in other modules

**Comment by alexey-pelykh** [comment]
@pedrobaeza changes made, please re-review

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py (hunk 2 above)
Comment (pedrobaeza) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py]: Exactly: the warning exactly for this.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**